### ISSUE TITLE
Create a rclone user in Docker image

### ISSUE THREAD

**Issue by cynful** [rclone/rclone#4831]
#### What is your current rclone version (output from `rclone version`)?
1.53.3

#### What problem are you are trying to solve?
Running as non-root user: rclone, more secure

#### How do you think rclone should be changed to solve that?
Update Dockerfile
```
RUN addgroup -g 1009 rclone && adduser -u 1009 -Ds /bin/sh -G rclone rclone
USER rclone
```
something like ^

thoughts? comments?

**Comment by ncw** [comment]
In the [install docs](https://rclone.org/install/#install-with-docker) we recommend using `--user $(id -u):$(id -g)`

Do you think setting the user to be non-root will break any existing workflows?

**Comment by robertbaker** [comment]
@ncw the issue I ran into with the docker image is that it doesn't do a mkdir /data with a chown on the USER. So for it to work, you can't use a named docker volume (not bind mount) rootless.

**Comment by ncw** [comment]
> the issue I ran into with the docker image is that it doesn't do a mkdir /data with a chown on the USER. So for it to work, you can't use a named docker volume (not bind mount) rootless.

I see.

I guess that is fixable though - do the chown during container startup? Or maybe it isn't?

> Do you think setting the user to be non-root will break any existing workflows?

This is the more important question though!

**Comment by robertbaker** [comment]
That would be hard to say.

An easy fix would be to declare /data in a VOLUMES directive.

**Comment by ncw** [comment]
> An easy fix would be to declare /data in a VOLUMES directive.

What would the consequence of that be if you didn't supply the --volume flag? Would it carry on writing to /data in the ephemeral filing system?

I read the docs here: https://docs.docker.comengine/reference/builder/#volume and I wasn't clear!

**Comment by darthShadow** [comment]
Yes, reference here: https://stackoverflow.com/a/46992367

**Comment by Charles546** [comment]
> In the [install docs](https://rclone.org/install/#install-with-docker) we recommend using `--user $(id -u):$(id -g)`
> 
> Do you think setting the user to be non-root will break any existing workflows?

It is preferred to have an user defined within the docker image, so it doesn't rely on the host user. This is important especially in managed kubernetes environments such as GKE. You want to run the image with a non-root user as a security best practice, but you can't assume there is an user on the host you can leverage.

**Comment by ncw** [comment]
> > In the [install docs](https://rclone.org/install/#install-with-docker) we recommend using `--user $(id -u):$(id -g)`
> > Do you think setting the user to be non-root will break any existing workflows?
> 
> It is preferred to have an user defined within the docker image, so it doesn't rely on the host user. This is important especially in managed kubernetes environments such as GKE. You want to run the image with a non-root user as a security best practice, but you can't assume there is an user on the host you can leverage.

Perhaps I'm mis-understanding, but using `--user $(id -u):$(id -g)` passes in numeric IDs which doesn't care whether there is a user on the host or not.

I'm happy to make the change so that we run as a non-privileged user by default, but I need someone who knows about such things to work out whether that is going to break people's workflows or not.

The `VOLUMES` thing above seems like an easy win and 100% backwards compatible if someone wants to send a PR with that.

**Comment by Charles546** [comment]
> I'm happy to make the change so that we run as a non-privileged user by default, but I need someone who knows about such things to work out whether that is going to break people's workflows or not.

It will, most likely, break some people's workflow. And, using `--user 0:0` might be a good workaround for the workflows broken by the change.

**Comment by ivandeex** [comment]
Closed by #5017
